src/python/pants/backend/debian/rules.py

```diff
@@ -1,64 +1,89 @@
-# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
-
 from dataclasses import dataclass
+from pathlib import PurePath
 
 from pants.backend.debian.target_types import (
-    DebianControlFile,
     DebianInstallPrefix,
     DebianPackageDependencies,
+    DebianSources,
 )
 from pants.core.goals.package import (
     BuiltPackage,
     BuiltPackageArtifact,
     OutputPathField,
     PackageFieldSet,
 )
-from pants.core.util_rules.system_binaries import BinaryPathRequest, BinaryPaths
+from pants.core.util_rules.system_binaries import BinaryPathRequest, BinaryPaths, TarBinary
+from pants.engine.fs import CreateDigest, DigestEntries, FileEntry
+from pants.engine.internals.native_engine import Digest
 from pants.engine.internals.selectors import Get
-from pants.engine.process import Process, ProcessResult
+from pants.engine.process import Process, ProcessCacheScope, ProcessResult
 from pants.engine.rules import collect_rules, rule
+from pants.engine.target import HydratedSources, HydrateSourcesRequest
 from pants.engine.unions import UnionRule
 from pants.util.logging import LogLevel
 
 
 @dataclass(frozen=True)
 class DebianPackageFieldSet(PackageFieldSet):
-    required_fields = (DebianControlFile, DebianInstallPrefix, DebianPackageDependencies)
+    required_fields = (DebianSources, DebianInstallPrefix, DebianPackageDependencies)
 
-    control: DebianControlFile
+    sources_dir: DebianSources
     install_prefix: DebianInstallPrefix
     packages: DebianPackageDependencies
     output_path: OutputPathField
 
 
-@rule(level=LogLevel.DEBUG)
-async def package_debian_package(field_set: DebianPackageFieldSet) -> BuiltPackage:
+@rule(level=LogLevel.INFO)
+async def package_debian_package(
+    field_set: DebianPackageFieldSet, tar_binary_path: TarBinary
+) -> BuiltPackage:
     dpkg_deb_path = await Get(
         BinaryPaths,
         BinaryPathRequest(
-            binary_name="touch",
+            binary_name="dpkg-deb",
             search_path=["/usr/bin"],
         ),
     )
     if not dpkg_deb_path.first_path:
-        raise OSError("Could not find the `touch` program on search paths ")
+        raise OSError(f"Could not find the `{dpkg_deb_path.binary_name}` program on search paths ")
+
+    hydrated_sources = await Get(HydratedSources, HydrateSourcesRequest(field_set.sources_dir))
 
-    output_filename = field_set.output_path.value_or_default(file_ending="deb")
+    # since all the sources are coming only from a single directory, it is
+    # safe to pick an arbitrary file and get its root directory name
```

Comment:
Nit: our style guide wants comments to be full sentences, i.e. w/ capitalization and a period: https://www.pantsbuild.org/docs/style-guide#style. That applies to a few comments in this PR

Reply:
Done!